go/vt/sqlparser/sql.y

```diff
@@ -1320,6 +1320,12 @@ float_length_opt:
   {
     $$ = LengthScaleOption{}
   }
+| '(' INTEGRAL ')'
```

Comment:
`float_length_opt` is a duplicate of `decimal_length_opt` with this, making the addition redundant, so we can use `decimal_length_opt` instead. It also allows `DOUBLE(n)`, which is not supported.